sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -654,6 +654,33 @@ class CastSuite extends CastSuiteBase {
       }
   }
 
+  test("SPARK-35768: Take into account year-month interval fields in cast") {
+    Seq(("1-1", YearMonthIntervalType(YEAR, YEAR), 12, 12, 12),
+      ("1-1", YearMonthIntervalType(YEAR, MONTH), 13, 12, 13),
+      ("1-1", YearMonthIntervalType(MONTH, MONTH), 13, 12, 13),
+      ("-1-1", YearMonthIntervalType(YEAR, YEAR), -12, -12, -12),
+      ("-1-1", YearMonthIntervalType(YEAR, MONTH), -13, -12, -13),
+      ("-1-1", YearMonthIntervalType(MONTH, MONTH), -13, -12, -13))
+      .foreach { case (str, dataType, ym, year, month) =>
+        checkEvaluation(cast(Literal.create(str), dataType), ym)
+        checkEvaluation(cast(Literal.create(s"INTERVAL '$str' YEAR TO MONTH"), dataType), ym)
+        checkEvaluation(cast(Literal.create(s"INTERVAL -'$str' YEAR TO MONTH"), dataType), -ym)
+        checkEvaluation(cast(Literal.create(s"INTERVAL '$str' YEAR"), dataType), year)
```

Comment:
The input interval string `INTERVAL '1-1' YEAR` is incorrect. I guess we must fail here. cc @cloud-fan @srielau

Reply:
Done